src/System.Reflection.DispatchProxy/src/System/Reflection/DispatchProxyGenerator.cs

```diff
@@ -426,13 +426,68 @@ internal void AddInterfaceImpl(Type iface)
                 _assembly.EnsureTypeIsVisible(iface);
 
                 _tb.AddInterfaceImplementation(iface);
+
+                // AccessorMethods -> Metadata mappings.
+                var propertyMap = new Dictionary<MethodInfo, PropertyAccessorInfo>();
+                var eventMap = new Dictionary<MethodInfo, EventAccessorInfo>();
+                foreach (var pi in iface.GetRuntimeProperties())
+                {
+                    var ai = new PropertyAccessorInfo(pi.GetMethod, pi.SetMethod);
+                    if (pi.GetMethod != null) propertyMap[pi.GetMethod] = ai;
+                    if (pi.SetMethod != null) propertyMap[pi.SetMethod] = ai;
+                }
+
+                foreach (var ei in iface.GetRuntimeEvents())
+                {
+                    var ai = new EventAccessorInfo(ei.AddMethod, ei.RemoveMethod);
+                    if (ei.AddMethod != null) eventMap[ei.AddMethod] = ai;
+                    if (ei.RemoveMethod != null) eventMap[ei.RemoveMethod] = ai;
+                }
+
                 foreach (MethodInfo mi in iface.GetRuntimeMethods())
                 {
-                    AddMethodImpl(mi);
+                    var mdb = AddMethodImpl(mi);
+                    PropertyAccessorInfo associatedProperty;
+                    if (propertyMap.TryGetValue(mi, out associatedProperty))
+                    {
+                        if (associatedProperty.InterfaceGetMethod == mi)
```

Comment:
See earlier comment about the hazards of using Equals (or "==") to compare MethodInfo's. It doesn't always work the way you want.